src/TEC/Plugin.php

```diff
@@ -115,14 +115,30 @@ public function register(): void {
 		// Run the label hooks
 		add_action( 'plugins_loaded', [ $this, 'init_label_hooks' ] );
 
+		// Start binds before WP All Import starts the import.
+		if ( ! has_action( 'pmxi_before_post_import', [ $this, 'init_import_hooks' ] ) ) {
+			add_action( 'pmxi_before_post_import', [ $this, 'init_import_hooks' ] );
```

Comment:
`add_action( 'pmxi_before_post_import', [ $this, 'init_import_hooks' ] );`

I messed up and targeted the wrong hook. It was `pmxi_before_xml_import` (which apparently is just loading the import configuration) and now I switched to `pmxi_before_post_import` (which should run prior to all imports).

@andrasguseo just an FYI, it was the wrong hook. We definitely do not want these hooks running always, they need to be targeted or we could create bugs or unintended side effects in other application behavior. Probably not major with what we have in here now, but it's always better to be careful and contain unintended side effects.

Thanks!